lib/token.js

```diff
@@ -0,0 +1,45 @@
+'use strict';
+
+const BaseModel = require('./base');
+const nodeify = require('./nodeify');
+const iron = require('iron');
+
+// Symbols for private members
+const password = Symbol('password');
+
+class TokenModel extends BaseModel {
+    /**
+     * Construct a TokenModel object
+     * @method constructor
+     * @param  {Object}    config
+     * @param  {Object}    config.datastore         Object that will perform operations on the datastore
+     * @param  {Number}    config.userId            The ID of the associated user
+     * @param  {String}    config.value             The token value
+     * @param  {String}    config.tokenId           The token ID
+     * @param  {String}    config.name              The token name
+     * @param  {String}    config.description       The token description
+     * @param  {Date}      config.lastUsed          The last time the token was accessed
+     * @param  {String}    config.password          The encryption password
+     */
+    constructor(config) {
+        super('token', config);
+        this[password] = config.password;
+    }
+
+    /**
+     * Update a token with sealed token value
+     * @method update
+     * @return {Promise}
+     */
+    update() {
+        return nodeify.withContext(iron, 'seal', [this.value, this[password], iron.defaults])
```

Comment:
I've been thinking that the way to solve this issue is to write custom getters and setters for the secret values. Then consumers of the model don't need to care about the seal/unseal operations (including things like get/update/save) because the actual value in the model is always the sealed one.

The drawback to this approach is that we may need to refactor places that already use these models to accept a promise as the return value instead of the raw value.

Something like:
```
get secret() {
  return nodeify.withContext(iron, 'unseal', [this.value, this[password], iron.defaults])
}
set secret() {
  // mark field as "dirty", then
  return nodeify.withContext(iron, 'seal', [this.value, this[password], iron.defaults])
}
```